marimo/_data/series.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+import datetime
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Union
+
+from marimo._dependencies.dependencies import DependencyManager
+
+if TYPE_CHECKING:
+    import pandas as pd
+    import polars as pl
+
+DataFrameSeries = Union["pd.Series[Any]", "pl.Series"]
+
+
+@dataclass
+class NumberSeriesInfo:
+    """
+    Represents a summary of a numeric series.
+    """
+
+    min: float
+    max: float
+    label: str
+
+
+@dataclass
+class CategorySeriesInfo:
+    """
+    Represents a summary of a categorical series.
+    """
+
+    categories: list[str]
+    label: str
+
+
+@dataclass
+class DateSeriesInfo:
+    """
+    Represents a summary of a date series.
+    """
+
+    min: str
+    max: str
+    label: str
+
+
+def get_number_series_info(series: Any) -> NumberSeriesInfo:
+    """
+    Get the summary of a numeric series.
+    """
+
+    def validate_number(value: Any) -> float:
+        value = float(value)
+        if not isinstance(value, (int, float)):
+            raise ValueError("Expected a number. Got: " + str(type(value)))
+        return value
+
+    if DependencyManager.has_pandas():
+        import pandas as pd
+
+        if isinstance(series, pd.Series):
+            return NumberSeriesInfo(
+                min=validate_number(series.min()),
+                max=validate_number(series.max()),
+                label=str(series.name),
```

Comment:
here and elsewhere, this should be `str(series.name) if series.name is not None else None`; otherwise, `None` gets printed as the label